common/xml.go

```diff
@@ -18,11 +23,93 @@ const (
 var (
 	//nolint:gochecknoglobals
 	validate          = validator.New()
+	ErrNoXMLRoot      = errors.New("xml document has no root")
 	ErrNotXMLChildren = errors.New("children must be of type 'XMLData' or 'XMLString'")
 	ErrNoSelfClosing  = errors.New("selfClosing cannot be true if children are not present")
 	ErrNoParens       = errors.New("value cannot contain < or >")
 )
 
+// XMLHTTPClient speaks from http client in XML.
```

Comment:
This comment line is a bit awkward sounding, do you mean something like "XMLHTTPClient is an HTTP client that can parse XML responses"?